### ISSUE TITLE
Distributed tracing: talk with jaeger via gRPC

### ISSUE THREAD

**Issue by flavio** [kubewarden/policy-server#78]
By default the opentelemetry-jaeger library talks with the Jaeger collector using UDP. UPD packets have a fixed size, if a trace is bigger than that it will simply be dropped.

All the current trace events generated by policy-server do not fit into a single UDP packet. As a result of that, no trace is going to be received by the Jaeger collector.

We should instead communicate with Jaeger over gRPC, which can fragment trace events over multiple packets.

**Comment by flavio** [comment]
A clarification about the UDP packet size: the size of the packet can be configured **but**:
  * The senders and collectors must agree on the max packet size to use, otherwise some traces will be lost
  * The packet size must be compatible with the limit set by the underlying OS.
  * Based on personal memory - hence I could be wrong - the max UDP packet size must be configured both at the OS level and then inside of the containerized workloads that want to take advantage of that

Long story short: there are many changes to mess it up with that. Based on my reading, the UDP transmission protocol is used by default because that provides the best performance. The Jaeger developers, when talking about that, hint that production deployments should always use a Jaeger collector running as a sidecar.
In that scenario the application will talk via UDP over a local connection.
